File: python/saddle-points/saddle_points.py
def saddle_points(matrix):
    for row in matrix:
            if len(row) != len(matrix[-1]):
                raise ValueError("matrix must be regular")

    result = [{'row': abscissa + 1, 'column': ordinate + 1} \
              for ordinate, col in enumerate(list(zip(*matrix))) \
              for abscissa, row in enumerate(matrix) \
              if min(col) == max(row)]
    return [{}] if len(result) == 0 else result

File: ruby/saddle-points/saddle_points.rb
class Matrix
  def initialize(matrix)
    @matrix = matrix
  end

  def rows
    @matrix.each_line.map do |line|
      line.scan(/\d+/).map(&:to_i)
    end
  end

  def columns
    rows.transpose
  end

  def saddle_points
    [].tap do |result|
      rows.each_with_index do |row, abscissa|
        columns.each_with_index do |col, ordinate|
          result << [abscissa, ordinate] if saddle_points?(col, row)
        end
      end
    end
  end

  def saddle_points?(column, row)
    column.min == row.max
  end

end